Arjun/Warehouse Winforms Project/Warehouse/ManagementApp/frmPickProduct.cs

```diff
@@ -0,0 +1,295 @@
+            {
+                product.Availability = product.Availability - product.Quantity;
+            }
+
+            else
+            {
+                product.Quantity = product.Availability;
+                product.Availability = 0;
+                //need to figure out what to do when products are unavailable
+                //Probably there must be some message.
+            }
+        }
+
+
+        private static double FindDistance(string v, string w)
+        {
+            string x1 = v.Substring(0, v.IndexOf(","));
+            string y1 = v.Substring(v.IndexOf(",") + 1);
+
+            string x2 = w.Substring(0, w.IndexOf(","));
+            string y2 = w.Substring(w.IndexOf(",") + 1);
+
+            double tempX1 = Convert.ToInt32(x1);
+            double tempY1 = Convert.ToInt32(y1);
+
+            double tempX2 = Convert.ToInt32(x2);
+            double tempY2 = Convert.ToInt32(y2);
+
+            double tempDistance = Math.Sqrt(((tempX1 - tempX2) * (tempX1 - tempX2)) +
+                                            ((tempY1 - tempY2) * (tempY1 - tempY2)));
+            return tempDistance;
+        }
+
+        private void btnNext_Click(object sender, EventArgs e)
+        {
+            if (productListUpdated.Count == 1)
+            {
+                frmProductSummary prodSummary = new frmProductSummary();
+                product = prodSummary.ShowSummary(product);
+                Close();
+                return;
+            }
+
+            IdentifyTheNextProductAndContinue();
+        }
+
+        
+        List<string[]> tempList = new List<string[]>();
+        List<string[]> tempList2 = new List<string[]>();
+
+        private void IdentifyTheNextProductAndContinue()
+        {
+            leastDistanceIndex = Convert.ToInt32(CompareDistances(productListUpdated));
+
+            SetProductPropertiesAndDisplay(productListUpdated);
+
+            totalCost += product.Cost;
+
+            dba.UpdateProductData(product);
+
+            product.TotalCost = totalCost;
+            product.Count = productCount;
+
+
+            tempList.Clear();
+            tempList2.Clear();
+
+
+            tempList.Add(productListUpdated[leastDistanceIndex]);
+
+            productListUpdated.RemoveAt(leastDistanceIndex);
+            productListUpdated.RemoveAt(0);
+
+
+            foreach (var blabla in productListUpdated)
+            {
+                tempList2.Add(blabla);
+            }
+
+            productListUpdated.Clear();
+
+            productListUpdated.Add(tempList[0]);
+
+            foreach (var blabla in tempList2)
+            {
+                productListUpdated.Add(blabla);
+            }
+        }
+
+        private string CompareDistances(List<string[]> productListUpdated)
+        {
+            
+            List<string[]> comparisonList = CreateComparisonList(productListUpdated);
+
+            SortedDictionary<double, string> distanceComparison = new SortedDictionary<double, string>();

```

Comment:
Never use double or float as keys. Double and float lose precision on different systems and will make your app not run correctly